modules/ui/sections/background_list.js

```diff
@@ -164,14 +97,41 @@ export function uiSectionBackgroundList(context) {
         });
     }
 
-    function drawListItems(layerList, type, change, filter) {
-        var sources = context.background()
+    function categoryMapping(layer) {
+        var categoryWeights = {
+            'photo': categoryGroups.photo,
+            'historicphoto': categoryGroups.photo,
+            'map': categoryGroups.map,
+            'historicmap': categoryGroups.map,
+            'elevation': categoryGroups.map,
+            'osmbasedmap': categoryGroups.osm,
+            'qa': categoryGroups.osm,
+            'other': categoryGroups.other
+        };
+        var layerIdWeights = {
+            'none': categoryGroups.other,
+            'custom': categoryGroups.other
+        };
+        return layerIdWeights[layer && layer.id] ||
+               categoryWeights[layer && layer.category] ||
+               categoryGroups.photo; // default = photo
+    }
+
+    function updateSources() {
+        return context.background()
             .sources(context.map().extent(), context.map().zoom(), true)
+            .filter(source => !source.isHidden());
+    }
+
+    function drawListItems(layerList, change, filter) {
+        var sources = _sources
             .filter(filter)
             .sort(function(a, b) {
-                return a.best() && !b.best() ? -1
-                    : b.best() && !a.best() ? 1
-                    : d3_descending(a.area(), b.area()) || d3_ascending(a.name(), b.name()) || 0;
+                return d3_descending(a.best ? 1 : 0, b.best ? 1 : 0) ||
+                       d3_ascending(a.overlay ? 1 : 0, b.overlay ? 1 : 0) ||
+                       d3_ascending(Math.floor(Math.log(a.area())/3), Math.floor(Math.log(b.area())/3)) ||
+                       d3_descending(a.endDate, b.endDate) ||
+                       d3_ascending(a.name(), b.name()) || 0;
```

Comment:
No sorting criteria is explicitly labeled, other than the icon for global sources, so adding more criteria could make the list less intuitive in some places. Now that the list is grouped by category, the category that these extra sorting criteria affect most would be “Maps based on OpenStreetMap”, but sources in that category not to have `best` or `endDate` set. If this becomes a problem in the future, maybe could consider a toggle to more straightforwardly sort by name versus vintage versus scope.

Reply:
AFAICS, the OSM layers are basically the most unaffected layers, since most of them are available on a global scale (indicated by the icon), and then sorted just alphabetically.

I think where the new sorting is having the most impact is for the "photo" layers, where one typically only just has a handful up to a dozen or so non-global layers. I think that's still a reasonable amount and doesn't need a more sophisticated sorting IMHO. FWIW, not even JOSM has anything more powerful in this regard. :wink: